### PR TITLE
Implement the new JS 1.1 API

### PR DESCRIPTION
**Pull request by iuliaion**
I have manually tested this manually on local frontend and verified that it works both with the new and old API version

### CODE REVIEW

File: u2f-chrome-extension/u2f-api.js [hunk 1]
```diff
@@ -88,6 +92,19 @@ u2f.Response;
...
+/**
+ * Data object for a single sign request.
+ * @typedef {enum {BLUETOOTH_RADIO, BLUETOOTH_LOW_ENERGY, USB, NFC}}
+ */
+u2f.Transport;
+
+
+/**
+ * Data object for a single sign request.
+// * @typedef {sequence<u2f.Transport>}
```
Comment (juanlang) [inline u2f-chrome-extension/u2f-api.js]: Why is this line commented out? Shouldn't it be sequence?

File: u2f-chrome-extension/u2f-api.js [hunk 2]
```diff
@@ -349,22 +390,50 @@ u2f.responseHandler_ = function(message) {
 
 /**
  * Dispatches an array of sign requests to available U2F tokens.
- * @param {Array<u2f.SignRequest>} signRequests
+ * @param {string=} appId
+ * @param {string=} challenge
+ * @param {Array<u2f.RegisteredKey>} registeredKeys
+// * @param {sequence<u2f.RegisteredKey>} registeredKeys
```
Comment (juanlang) [inline u2f-chrome-extension/u2f-api.js]: Ditto.

File: u2f-chrome-extension/u2f-api.js [hunk 3]
```diff
@@ -17,6 +17,11 @@
  */
 var u2f = u2f || {};
 
+/**
+ * FIDO U2F Javascript API Version
+ * @number
+ */
+var JS_API_VERSION;
```
Comment (juanlang) [inline u2f-chrome-extension/u2f-api.js]: If this value can change, it shouldn't be CONST case.

File: u2f-chrome-extension/u2f-api.js [hunk 4]
```diff
@@ -349,22 +390,50 @@ u2f.responseHandler_ = function(message) {
...
-    var req = {
-      type: u2f.MessageTypes.U2F_SIGN_REQUEST,
-      signRequests: signRequests,
-      timeoutSeconds: (typeof opt_timeoutSeconds !== 'undefined' ?
-          opt_timeoutSeconds : u2f.EXTENSION_TIMEOUT_SEC),
-      requestId: reqId
-    };
-    port.postMessage(req);
+    var request;
+    if (JS_API_VERSION === undefined || JS_API_VERSION < 1.1) {
```
Comment (juanlang) [inline u2f-chrome-extension/u2f-api.js]: This is a little confusing. The value JS_API_VERSION gets set by calling getApiVersion(), but what if the caller of this function never calls that one? Should you call it implicitly here instead?

File: u2f-chrome-extension/u2f-api.js (hunk 1 above)
Comment (iuliaion) [inline u2f-chrome-extension/u2f-api.js]: As far as I know, there is no sequence construct in javascript. I was unsure about sequence vs. Array, as the previous API calls it Array and the new one sequence. I checked with Alexei and Nick and in the end concluded this should just be array. Then I forgot to delete the sequence line. What should it be?

File: u2f-chrome-extension/u2f-api.js (hunk 4 above)
Comment (iuliaion) [inline u2f-chrome-extension/u2f-api.js]: Good point. I was thinking just in the confines of this demo. But, yes, that makes a lot of sense if the extension is being used by other sites.

I changed to make the call to find out the extension version here.

File: u2f-chrome-extension/u2f-api.js (hunk 3 above)
Comment (iuliaion) [inline u2f-chrome-extension/u2f-api.js]: Thanks. Renamed it to js_api_version.

**Comment by juanlang** [comment]
lgtm

**Pull request merged**